changelog.d/15748.removal

```diff
@@ -0,0 +1 @@
+Remove experimental [MSC2716](https://github.com/matrix-org/matrix-spec-proposals/pull/2716) implementation to incrementally import history into existing rooms.
```

Comment:
Do we need to worry about [gracefully dropping database tables](https://github.com/matrix-org/synapse/blob/fcc3ca37e1b404981d9a0d6f2708e14407775b97/docs/development/database_schema.md) for an experimental feature like this?

Relevant tables to drop:

```sql
-- Drop tables related to MSC2716 since the implementation is being removed
DROP TABLE insertion_events;
DROP TABLE insertion_event_edges;
DROP TABLE insertion_event_extremities;
DROP TABLE batch_events;
```

Can we just stop writing to the tables in the Synapse version that this PR ships in, then in the Synapse version after, bump the `SCHEMA_COMPAT_VERSION` and drop the tables?

Reply:
I'd say we may as well? We still want to be careful that we don't accidentally break upgrades/downgrades for people that have turned on experimental features.

Easiest way of tracking the task of dropping those tables is to create an issue with the future maintenance label and add it to one of the milestones.

Reply:
Created https://github.com/matrix-org/synapse/issues/15786 to track this :fast_forward: